geocoder_tester/base.py

```diff
@@ -31,6 +60,41 @@ def __str__(self):
         return self.error
 
 
+class DuplicatesException(Exception):
+    """ custom exception for duplicates reporting. """
+
+    def __init__(self, duplicates, params):
+        super().__init__()
+        self.duplicates = duplicates
+        self.query = params.pop('q')
```

Comment:
`pop` may be confusing here: the params object is modified when raising the exception